### PR TITLE
Use language in Nets

### PR DESCRIPTION
**Pull request by Magnusrm**
## Description

Giva current selected language as query parameter to payment endpoints in order to make use of the language functionality in backend.

Also solved an insconsistency with going directly to Nets where it would only work the first time, since the response for paymentInformation would be cached and it would seem like a payment existed even though it had been cleaned up in backend.

## Related Issue(s)

- closes #{issue number}

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [x] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/features/payment/PaymentInformationProvider.tsx [hunk 1]
```diff
@@ -16,10 +17,12 @@ export function usePaymentInformationQueryDef(
   instanceId?: string,
 ): QueryDefinition<PaymentResponsePayload> {
   const { fetchPaymentInformation } = useAppQueries();
+  const selectedLanguage = useCurrentLanguage();
   return {
     queryKey: ['fetchPaymentInfo'],
-    queryFn: instanceId ? () => fetchPaymentInformation(instanceId) : skipToken,
+    queryFn: instanceId ? () => fetchPaymentInformation(instanceId, selectedLanguage) : skipToken,
     enabled: enabled && !!instanceId,
+    gcTime: 0,
```
Comment (bjosttveit) [inline src/features/payment/PaymentInformationProvider.tsx:25]: Good catch!

**Review by bjosttveit — APPROVED** [review]
Looks good, is it not then necessary to clear the order-details-query-cache when the language changes? (in the PaymentDetails component)

**Comment by Magnusrm** [comment]
@bjosttveit The `PaymentDetails` component uses the `orderDetailsProvider` and not the `PaymentInformationProvider` so it is sadly still needed in order to update orderlines dynamically.  

A solution we had earlier for the `PaymentDetails` component was to always refetch the orderlines after the datamodel saved data, regardless if the datafield was relevant or not. This allowed us to remove mapping, however it might be a bit much to allways refetch in regards to performance.

**Comment by bjosttveit** [comment]
I just meant that if the order details now depends on the language, maybe also clear the cache when the language changes, in addition to when the mapping changes. If that makes any difference that is 😄

**Pull request merged**